### PR TITLE
Support keyPrefix

### LINKED ISSUE #223: Feature key prefixes
Support key prefixes https://github.com/i18next/react-i18next/pull/1371
```tsx
const { t } = useTranslation('translation', { keyPrefix: 'very.deeply.nested' });
const text = t('key'); // extract as "very.deeply.nested.key"
```

### PR DESCRIPTION
**Pull request by BorisTB**
Support optional keyPrefix option

https://www.i18next.com/overview/api#getfixedt
https://react.i18next.com/latest/usetranslation-hook#optional-keyprefix-option

Resolves #223

### CODE REVIEW

File: src/keys.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ interface I18NextParsedOptions {
  */
 export interface ExtractedKey {
   key: string;
+  keyPrefix?: string;
```
Comment (gilbsgilbs) [inline src/keys.ts:21]: Would it be possible to move this option to `parsedOptions`, make it non-optional and default it to `null`?

File: src/keys.ts [hunk 2]
```diff
@@ -54,15 +55,27 @@ function parseExtractedKey(key: ExtractedKey, config: Config): TranslationKey {
     const nsSeparatorPos = cleanKey.indexOf(config.nsSeparator);
 
     if (nsSeparatorPos !== -1) {
+      if (key.keyPrefix) {
+        throw new Error(
```
Comment (gilbsgilbs) [inline src/keys.ts:59]: hat is i18next behavior when you use both? Does it error out or pick one of the two namespaces? I think we should align with their behavior.

In any case, we need a test for this.

File: src/keys.ts [hunk 3]
```diff
@@ -54,15 +55,27 @@ function parseExtractedKey(key: ExtractedKey, config: Config): TranslationKey {
...
+          `Do not use the keyPrefix option if you want to use keys with prefixed namespace notation.
+          key: ${cleanKey}
```
Comment (gilbsgilbs) [inline src/keys.ts:61]: Nit: indentation is not desired in an error message. Maybe just put the values inline, in parentheses.

File: src/keys.ts [hunk 4]
```diff
@@ -54,15 +55,27 @@ function parseExtractedKey(key: ExtractedKey, config: Config): TranslationKey {
...
+      }
+
       ns = cleanKey.slice(0, nsSeparatorPos);
       cleanKey = cleanKey.slice(nsSeparatorPos + 1);
     }
   }
 
   let keyPath = Array<string>();
   if (config.keySeparator) {
+    if (key.keyPrefix) {
```
Comment (gilbsgilbs) [inline src/keys.ts:73]: Is a key separator always needed? Doesn't i18next have a fallback or something?

**Review by gilbsgilbs — CHANGES_REQUESTED** [review]
Interesting contribution, thank you 👍. Nice tests also.

Just have a few minor remarks that may or may not require change on your end.

File: src/keys.ts (hunk 4 above)
Comment (BorisTB) [reply src/keys.ts:73]: It is always needed, without it we can't connect prefix with key

https://github.com/i18next/i18next/blob/a6c6680b151fd66e4b4cc2f443aaeaa5d057a453/src/i18next.js#L393-L399


i18next have fallback to `.` but i believe your lib does that too in `parseConfig`
https://github.com/gilbsgilbs/babel-plugin-i18next-extract/blob/5eb8f8bbfaa4f651a662c864f607db0aa455e69e/src/config.ts#L67

File: src/keys.ts (hunk 2 above)
Comment (BorisTB) [reply src/keys.ts:59]: I'm not really sure what will happend if i use both, they just have warning in documentation it won't work. 
https://www.i18next.com/overview/api#getfixedt

I'll test exact behaviour of i18next and try to make tests for it

**Comment by BorisTB** [comment]
PR updated

**Comment by gilbsgilbs** [comment]
Thanks!

**Pull request merged**